model2vec/train/README.md

```diff
@@ -0,0 +1,137 @@
+
+# Installation
+
+To train, make sure you install the training extra:
+
+```
+pip install model2vec[training]
+```
+
+# Quickstart
+
+To train a model, simply initialize it using a `StaticModel`, or from a pre-trained model, as follows:
+
+```python
+from model2vec.distill import distill
+from model2vec.train import StaticModelForClassification
+
+# From a distilled model
+distilled_model = distill("baai/bge-base-en-v1.5")
+classifier = StaticModelForClassification.from_static_model(distilled_model)
+
+# From a pre-trained model: potion is the default
+classifier = StaticModelForClassification.from_pretrained(model_name="minishlab/potion-base-8m")
+```
+
+This creates a very simple classifier: a StaticModel with a single 512-unit hidden layer on top. You can adjust the number of hidden layers and the number units through some parameters on both functions. Note that the default for `from_pretrained` is [potion-base-8m](https://huggingface.co/minishlab/potion-base-8M), our best model to date. This is our recommended path if you're working with general English data.
+
+Now that you have created the classifier, let's just train a model. The example below assumes you have the [`datasets`](https://github.com/huggingface/datasets) library installed.
+
+```python
+import numpy as np
+from datasets import load_dataset
+
+# Load the subj dataset
+ds = load_dataset("setfit/subj")
+train = ds["train"]
+test = ds["test"]
+
+s = perf_counter()
+classifier = classifier.fit(train["text"], train["label"])
+
+predicted = classifier.predict(test["text"])
+print(f"Training took {int(perf_counter() - s)} seconds.")
+# Training took 81 seconds
+accuracy = np.mean([x == y for x, y in zip(predicted, test["label"])]) * 100
+print(f"Achieved {accuracy} test accuracy")
+# Achieved 91.0 test accuracy
+```
+
+As you can see, we got a pretty nice 91% accuracy, with only 81 seconds of training.
+
+The training loop is handled by [`lightning`](https://pypi.org/project/lightning/). By default the training loop splits the data into a train and validation split, with 90% of the data being used for training and 10% for validation. By default, it runs with early stopping on the validation set accuracy, with a patience of 5.
+
+Note that this model is as fast as you're used to from us:
+
+```python
+from time import perf_counter
+
+s = perf_counter()
+classifier.predict(test["text"])
+print(f"Took {int((perf_counter() - s) * 1000)} milliseconds for {len(test)} instances on CPU.")
+# Took 67 milliseconds for 2000 instances on CPU.
+```
+
+# Persistence
+
+You can turn a classifier into a scikit-learn compatible pipeline, as follows:
+
+```python
+pipeline = classifier.to_pipeline()
+```
+
+This pipeline object can be persisted using standard pickle-based methods, such as [joblib](https://joblib.readthedocs.io/en/stable/). This makes it easy to use your model in inferene pipelines (no installing torch!).
```

Comment:
I believe there are some security concerns with pickle and joblib https://calmcode.io/course/scikit-save/security.

Reply:
We load using `skops` internally, so pushing to the hub and so on is safe if you use this library. But yeah I'll add a note to this effect.